frontend/src/container/GroupedNotificationItem.jsx

```diff
@@ -17,124 +17,150 @@ import {
   TracimComponent,
   formatAbsoluteDate
 } from 'tracim_frontend_lib'
-import { updateNotification } from '../action-creator.sync'
+import NotificationItem from '../component/NotificationItem.jsx'
+// import putNotificationAsRead from '../action-creator.async.js'
 import { escape as escapeHtml, uniqBy } from 'lodash'
 
-function getGroupedNotificationDetails (props, notification) {
-  const { getNotificationDetails, t } = props
-
-  const numberOfContributionTypes = uniqBy(notification.group, 'type').length
-  const numberOfWorkspaces = uniqBy(notification.group.map(notification => notification.workspace), 'id').length
-  const numberOfContents = uniqBy(notification.group.filter(notification => notification.content)
-    .map(notification => {
-      return notification.content.type === TLM_SUB.COMMENT
-        ? notification.content.parentId
-        : notification.content.id
-    })).length
-
-  let escapedAuthorList = ''
-  if (notification.author) {
-    if (notification.author.length === 1) {
-      escapedAuthorList = escapeHtml(notification.author[0].publicName)
-    } else if (notification.author.length === 2) {
-      escapedAuthorList = `${escapeHtml(notification.author[0].publicName)} ${t('and')} ${escapeHtml(notification.author[1].publicName)}`
-    } else {
-      escapedAuthorList = `${escapeHtml(notification.author[0].publicName)} ${t('and {{numberOfAuthors}} other people', { numberOfAuthors: notification.author.length - 1 })}`
+export const GroupedNotificationItem = props => {
+  const [notificationList, setNotificationList] = useState([])
+  const [isGrouped, setIsGrouped] = useState(true)
+
+  useEffect(() => {
+    const tmpNotificationList = []
+    props.groupedNotifications.group.forEach(notification => {
+      tmpNotificationList.push(notification)
+    })
+    setNotificationList(tmpNotificationList)
+  }, [props.groupedNotifications])
+
+  const getGroupedNotificationDetails = (notification) => {
```

Comment:
If I'm not wrong this function does not use any state, so you could move it outside the component definition which would help understand the global behavior of this component.